src/v/cluster/controller_backend.cc

```diff
@@ -476,6 +481,40 @@ ss::future<> controller_backend::fetch_deltas() {
       });
 }
 
+ss::future<> controller_backend::clear_orphan_topic_files() {
+    auto topics = _topics.local().all_topics();
+    storage::log_manager::ntp_to_revision topics_on_node;
+    // Init with default namespace to clean if there is no topics
+    absl::flat_hash_set<model::ns> namespaces = {{model::kafka_namespace}};
+    for (const auto& t : topics) {
+        namespaces.emplace(t.ns);
+        auto meta = _topics.local().get_topic_assignments(t);
+        if (meta) {
+            for (const auto& p : *meta) {
+                model::ntp ntp(t.ns, t.tp, p.id);
+                auto revision = _topics.local()
+                                  .get_topic_metadata(
+                                    model::topic_namespace_view(ntp))
+                                  ->get_revision();
+                auto previous_replica_set
+                  = _topics.local().get_previous_replica_set(ntp);
+                if (
+                  contains_node(_self, p.replicas)
+                  || (previous_replica_set && contains_node(_self, *previous_replica_set))) {
```

Comment:
Why do we care about the previous replica set?

Reply:
If movement is in progress, we shouldn't delete topic because we can cancel movement.

Reply:
Your answer makes sense, but it doesn't match the code (very likely it's just me not understanding). If there's a `previous_replica_set` it means that there's a move in progress, right? Still, we add the ntp to `topics_on_node` as a candidate for deletion. Doesn't this mean that we may delete during a move?

Reply:
topics_on_node is current topics_on_node that we don't want to delete.
It is not candidates. We delete topic files if it doesn't exist in topics_on_node

Reply:
Understood. Thank you!